src/query/storage/cache/redis_cache.go

```diff
@@ -36,14 +54,34 @@ const (
 	// BucketWindowGetOrFetch() will use this key, see function for more details
 	BucketKeyPrefix = "sliding_window"
 
+	// Number of buckets that are allowed to be missing before we just default to M3DB entirely
 	AllowedMissingBuckets = 1
+
+	// The time that needs to pass before we can set a bucket (to make sure data is fully filled in)
+	SetBufferTime = 1 * time.Minute
```

Comment:
What is this used for?

Reply:
It is the time difference that needs to be present between the end of the query in time and the bucket end in order to set the bucket in Redis. This is to make sure when we set the bucket that it is fully loaded in on the M3DB side